app/api/entities/validation/validateMetadataField.ts

```diff
@@ -122,6 +123,41 @@ const validateRelationshipForeignIds = async (
   return [];
 };
 
+const validateSameRelationshipsMatch = (
+  property: PropertySchema,
+  entity: EntitySchema,
+  template: TemplateSchema,
+  value: MetadataObjectSchema[] | undefined
+) => {
+  if (property.type !== propertyTypes.relationship) {
+    return [];
+  }
+
+  const sameProps =
+    template.properties?.filter(
+      p =>
+        p.type === propertyTypes.relationship &&
+        p.content?.toString() === property.content?.toString() &&
+        p.relationType?.toString() === property.relationType?.toString()
+    ) || [];
+
+  const valid = Boolean(
+    sameProps.every(
+      p => !entity.metadata || JSON.stringify(entity.metadata[p.name]) === JSON.stringify(value)
```

Comment:
Is there a chance for the metadata values to be the same but be in different order, making this stringify comparison fail?

Reply:
there is a new implementation